Library/src/Systems/Assets/AssetSystem.cpp

```diff
@@ -847,4 +866,303 @@ CSP_ASYNC_RESULT_WITH_PROGRESS void
+			AssetData.SetMimeType("application/json");
+			AssetData.Buffer	   = const_cast<char*>(MaterialJson.c_str());
+			AssetData.BufferLength = MaterialJson.Length();
+
+			UploadAssetData(CreatedAssetCollection, CreatedAsset, AssetData, UploadMaterialCallback);
+		};
+
+		const csp::common::String MaterialAssetName = MATERIAL_ASSET_NAME_PREFIX + SpaceId + "_" + Name;
+
+		CreateAsset(CreateAssetCollectionResult.GetAssetCollection(), MaterialAssetName, nullptr, nullptr, EAssetType::MATERIAL, CreateAssetCB);
+	};
+
+	const csp::common::String MaterialCollectionName = MATERIAL_ASSET_COLLECTION_NAME_PREFIX + SpaceId + "_" + Name;
+
+	CreateAssetCollection(SpaceId, nullptr, MaterialCollectionName, nullptr, EAssetCollectionType::DEFAULT, nullptr, CreateAssetCollectionCB);
+}
+
+void AssetSystem::UpdateMaterial(const GLTFMaterial& Material, NullResultCallback Callback)
+{
+	// 1. Get asset collection
+	auto GetAssetCollectionCB = [this, Material, Callback](const AssetCollectionResult& CreateAssetCollectionResult)
+	{
+		if (CreateAssetCollectionResult.GetResultCode() != EResultCode::Success)
+		{
+			Callback(NullResult(CreateAssetCollectionResult.GetResultCode(), CreateAssetCollectionResult.GetHttpResultCode()));
+			return;
+		}
+
+		// 2. Get asset
+		const AssetCollection& CreatedAssetCollection = CreateAssetCollectionResult.GetAssetCollection();
+
+		auto GetAssetCB = [this, Callback, Material, CreatedAssetCollection](const AssetResult& CreateAssetResult)
+		{
+			if (CreateAssetResult.GetResultCode() != EResultCode::Success)
+			{
+				Callback(NullResult(CreateAssetResult.GetResultCode(), CreateAssetResult.GetHttpResultCode()));
+				return;
+			}
+
+			// 3. Upload material
+			auto UploadMaterialCallback = [this, Callback, Material](const UriResult& UploadResult)
+			{
+				Callback(NullResult(UploadResult.GetResultCode(), UploadResult.GetHttpResultCode()));
+			};
+
+			csp::common::String MaterialJson = json::JsonSerializer::Serialize(Material);
+			const Asset& CreatedAsset		 = CreateAssetResult.GetAsset();
+
+			BufferAssetDataSource AssetData;
+			AssetData.SetMimeType("application/json");
+			AssetData.Buffer	   = const_cast<char*>(MaterialJson.c_str());
+			AssetData.BufferLength = MaterialJson.Length();
+
+			UploadAssetData(CreatedAssetCollection, CreatedAsset, AssetData, UploadMaterialCallback);
+		};
+
+		GetAssetById(Material.GetAssetCollectionId(), Material.GetAssetId(), GetAssetCB);
+	};
+
+	GetAssetCollectionById(Material.GetAssetCollectionId(), GetAssetCollectionCB);
+}
+
+void AssetSystem::DeleteMaterial(const GLTFMaterial& Material, NullResultCallback Callback)
+{
+	// 1. Delete asset
+	auto DeleteAssetCB = [this, Callback, &Material](const NullResult& DeleteAssetResult)
+	{
+		if (DeleteAssetResult.GetResultCode() != EResultCode::Success)
+		{
+			Callback(NullResult(DeleteAssetResult.GetResultCode(), DeleteAssetResult.GetHttpResultCode()));
+			return;
+		}
+
+		// 2. Deletion asset collection
```

Comment:
[Nit] all your other code comments start with a verb, so this one would be "Delete asset collection".